src/Radicle/Daemon/HttpApi.hs

```diff
@@ -39,51 +77,71 @@ instance FromHttpApiData MachineId where
 instance ToHttpApiData MachineId where
     toUrlPiece (MachineId id) = id
 
-newtype QueryRequest = QueryRequest { expression :: Value }
-  deriving (Generic)
+newtype QueryRequestF a = QueryRequest { expression :: a }
+  deriving (Functor, Foldable, Traversable, Generic, A.ToJSON, A.FromJSON)
+
+type QueryRequest = QueryRequestF Value
 
-instance A.ToJSON QueryRequest where
+instance {-# OVERLAPPING #-} A.ToJSON QueryRequest where
     toJSON QueryRequest{..} = A.object
         [ "expression" .= valueToJson expression
```

Comment:
It seems a little ugly to have two different ways of encoding to JSON, and then picking one arbitrarily as the instance...